modules/common/src/main/scala/surge/internal/akka/ActorWithTracing.scala

```diff
@@ -3,29 +3,75 @@
 package surge.internal.akka
 
 import akka.AroundReceiveActor
-import io.opentracing.Scope
-import surge.internal.utils.SpanSupport
-import surge.tracing.TracedMessage
+import akka.actor.ActorRef
+import io.opentracing._
+import surge.tracing.{ SpanExtensions, TracePropagation, TracedMessage }
 
-trait ActorWithTracing extends AroundReceiveActor with SpanSupport {
-  private var activeScope: Option[Scope] = None
+trait ActorWithTracing extends AroundReceiveActor with ActorOps with SpanExtensions {
 
-  def tracedMessage[T](msg: T): TracedMessage[T] = {
-    TracedMessage(tracer, msg, tracer.activeSpan())
+  type MessageNameExtractor = PartialFunction[Any, String]
+
+  protected var activeSpan: Span = _
+
+  implicit val tracer: Tracer
+
+  private def actorClassFullName: String = this.getClass.getName
+
+  private def actorClassSimpleName: String = this.getClass.getSimpleName
+
+  def messageNameForTracedMessages: MessageNameExtractor = PartialFunction.empty[Any, String]
+
+  private def getMessageName(msg: Any): String = {
+    if (messageNameForTracedMessages.isDefinedAt(msg)) {
+      messageNameForTracedMessages(msg)
+    } else {
+      msg.getClass.getSimpleName
+    }
+  }
+
+  private def getFields: Map[String, String] = {
+    Map(
+      "actor" -> actorClassSimpleName,
+      "actor (fqcn)" -> actorClassFullName,
+      "receiver path" -> self.prettyPrintPath,
+      "sender path" -> sender().prettyPrintPath)
   }
 
   override def doAroundReceive(receive: Receive, msg: Any): Unit = {
     msg match {
-      case traced: TracedMessage[_] =>
-        activeScope = Some(tracer.activateSpan(traced.activeSpan(tracer)))
-        superAroundReceive(receive, traced.message)
-      case _ =>
+      case tracedMsg: TracedMessage[_] =>
+        val messageName: String = getMessageName(tracedMsg.message)
+        val operationName: String = s"$actorClassSimpleName:$messageName"
+        val newSpan: Span = TracePropagation.childFrom(tracedMsg, operationName)
+        activeSpan = newSpan
+        val fields: Map[String, String] = getFields +
+          ("message" -> messageName) +
+          ("message (fqcn)" -> tracedMsg.message.getClass.getName)
+        newSpan.log("receive", fields)
+        superAroundReceive(receive, tracedMsg.message)
+      case other =>
+        val messageName: String = other.getClass.getSimpleName
```

Comment:
Any reason not to do a `getMessageName` here as well?

Reply:
good point, not a bad idea to do a getMessageName even for messages that are not TracedMessages ...

Reply:
done ;-) also removed some duplication